sc/src/main/java/uk/ac/ebi/atlas/experimentpage/ExperimentPageContentService.java

```diff
@@ -0,0 +1,142 @@
+
+    private static final Gson gson = new Gson();
+
+    public ExperimentPageContentService(ExperimentFileLocationService experimentFileLocationService,
+                                        DataFileHub dataFileHub,
+                                        SingleCellContentService singleCellContentService) {
+        this.experimentFileLocationService = experimentFileLocationService;
+        this.dataFileHub = dataFileHub;
+        this.singleCellContentService = singleCellContentService;
+    }
+
+    public JsonObject getTsnePlotDataAsJson() {
+        JsonObject result = new JsonObject();
+
+        JsonArray availableClusters = new JsonArray();
+        Arrays.stream(new int[] {2, 3, 4, 5, 6, 7, 8, 9, 10}).forEach(availableClusters::add);
+        result.add("ks", availableClusters);
+
+        // TODO Get available perplexities from scxa_tsne https://www.pivotaltracker.com/story/show/154898174
+        JsonArray perplexityArray = new JsonArray();
+        Arrays.stream(new int[] {1, 5, 10, 15, 20}).forEach(perplexityArray::add);
+        result.add("perplexities", perplexityArray);
+
+        JsonArray units = new JsonArray();
+        units.add("TPM");
+
+        result.add("units", units);
+
+        result.addProperty("suggesterEndpoint", "json/suggestions");
+
+        return result;
+    }
+
+    public JsonObject getExperimentDesignAsJson(String experimentAccession, JsonObject experimentDesignTableAsJson, String accessKey) {
+        JsonObject result = new JsonObject();
+
+        result.add("table", experimentDesignTableAsJson);
+
+        String fileUri = experimentFileLocationService.getFileUri(experimentAccession, ExperimentFileType.EXPERIMENT_DESIGN, accessKey).toString();
+        result.addProperty("downloadUrl", fileUri);
+
+        return result;
+    }
+
+    public JsonArray getDownloadsAsJson(String experimentAccession, String accessKey) {
+        JsonArray result = new JsonArray();
+
+        result.add(getExperimentFileAsJson(ExperimentFileType.SDRF, experimentAccession, accessKey));
+        result.add(getExperimentFileAsJson(ExperimentFileType.CLUSTERING, experimentAccession, accessKey));
+        result.add(getExperimentFileAsJson(ExperimentFileType.EXPERIMENT_DESIGN, experimentAccession, accessKey));
+        result.add(getExperimentFileAsJson(ExperimentFileType.QUANTIFICATION_FILTERED, experimentAccession, accessKey));
+
+        return result;
+    }
+
+    public JsonArray getAnalysisMethodsAsJson(String experimentAccession) {
+        JsonArray result;
+
+        try (TsvStreamer tsvStreamer =
+                     dataFileHub.getSingleCellExperimentFiles(experimentAccession).softwareUsed.get()) {
+                            result = gson.toJsonTree(tsvStreamer.get().collect(Collectors.toList())).getAsJsonArray();
+        };
+
+        return result;
+    }
+
+    public JsonArray getResourcesAsJson(String experimentAccesssion, String accessKey, ExternallyAvailableContent.ContentType contentType) {
```

Comment:
This was used to retrieve the link to Array Express, which I have since removed. It's the same way it's done in Expression Atlas. I did wonder if I should remove it, because when we do implement a link to <insert service here>, I would refactor the whole thing anyway...